### PR TITLE
docs(core): nx cloud workflows intro

### PR DESCRIPTION
**Pull request by isaacplmann**
Adds a page describing [nx cloud workflows](https://nx-dev-git-fork-isaacplmann-docs-nx-cloud-workflows-nrwl.vercel.app/nx-cloud/intro/nx-cloud-workflows)

### CODE REVIEW

File: docs/nx-cloud/intro/nx-cloud-workflows.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# Nx Cloud Workflows
+
+{% youtube
+src="https://www.youtube.com/embed/IQ5YyEYZw68?start=3371"
+title="Introducing Nx Cloud Workflows"
+width="100%" /%}
+
+## Powerful CI Capabilities Optimized for Nx monorepos
```
Comment (juristr) [inline docs/nx-cloud/intro/nx-cloud-workflows.md:8]: I'm missing a bit the "Why" in this and next section. We describe what Nx Cloud workflow does etc, but not why you want to use it. We still need the latter (e.g. evolution of DTE, automatic scaling etc) but I think if we describe it from the "why beneficial" standpoint it resonates better

Things like
- "legacy" CI is low-level command execution on machines in the cloud
- they don't understand your workspace, just run commands
- Nx Cloud Workflows is task centric, you give it tasks to run and it knows how to split them into lots of fine-grained commands/targets based on your workspace structure
- cost savings due to automated DTE scaling, which before/right now you have to figure out almost on your own; like what number of agents makes most sense at your current monorepo situation to have good parallelism while still not wasting money by spinning up machines

something like this

**Review by juristr — CHANGES_REQUESTED** [review]

File: docs/nx-cloud/intro/nx-cloud-workflows.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Nx Cloud Workflows
+
+{% youtube
```
[the remark this answers is not in the document]
Comment (isaacplmann) [reply docs/nx-cloud/intro/nx-cloud-workflows.md:3]: waiting on a youtube link

File: docs/nx-cloud/intro/nx-cloud-workflows.md [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      npx nx-cloud start-agent
+```
+
+## Security
+
+You have the choice between an on-premise setup or using the hosted version of Nx Cloud Workflows. The hosted version of Nx Cloud Workflows uses gVisor to ensure that each container is sandboxed in a virtualized environment so your build system is protected from attackers. gVisor is the same tool that is used to protect Google Cloud Functions.
+
+## Early Access
+
+If you are interested in being one of the first organizations to experience Nx Cloud Workflows, contact [cloud-support@nrwl.io](mailto://cloud-support@nrwl.io).
```
Comment (isaacplmann) [inline docs/nx-cloud/intro/nx-cloud-workflows.md:56]: I'll add the sign up form if it is published before this PR is merged

**Review by juristr — APPROVED** [review]

File: docs/nx-cloud/intro/nx-cloud-workflows.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (isaacplmann) [reply docs/nx-cloud/intro/nx-cloud-workflows.md]: removed the video.  I'll add it in a separate PR

**Pull request merged**